test/SPIRV/builtin_vars-decorate.ll

```diff
@@ -0,0 +1,78 @@
+; RUN: llvm-as < %s | llvm-spirv -spirv-text -o %t
+; RUN: FileCheck < %t %s
+
+; ModuleID = 'test.cl'
+target datalayout = "e-p:32:32-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024"
+target triple = "spir-unknown-unknown"
+
+; CHECK: 4 Decorate [[GID:[0-9]*]] BuiltIn 24

```

Comment:
Here you define variable GID, which not used later, moreover it's re-defined.
I recommend either:
1. Do you not use variables: CHECK: 4 Decorate {{[0-9]+}} BuiltIn 24
2. Improve check and validate that Decoration is applied to correct variable. You can check it by validating data type of the variable.